src/transport/CertificateAuthenticatedSessionEstablishment.h

```diff
@@ -0,0 +1,175 @@
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines the CHIP CASE Session object that provides
+ *      APIs for constructing secure session using certificate from device's
+ *      operational credentials.
+ */
+
+#pragma once
+
+#include <crypto/CHIPCryptoPAL.h>
+#include <protocols/secure_channel/Constants.h>
+#include <support/Base64.h>
+#include <system/SystemPacketBuffer.h>
+#include <transport/AuthenticatedSessionEstablishmentDelegate.h>
+#include <transport/PeerConnectionState.h>
+#include <transport/SecureSession.h>
+#include <transport/raw/MessageHeader.h>
+#include <transport/raw/PeerAddress.h>
+
+namespace chip {
+
+using namespace Crypto;
+
+class DLL_EXPORT CertificateAuthenticatedSessionEstablishment
+{
+public:
+    CertificateAuthenticatedSessionEstablishment();
+    CertificateAuthenticatedSessionEstablishment(CertificateAuthenticatedSessionEstablishment &&)      = default;
+    CertificateAuthenticatedSessionEstablishment(const CertificateAuthenticatedSessionEstablishment &) = default;
+    CertificateAuthenticatedSessionEstablishment & operator=(const CertificateAuthenticatedSessionEstablishment &) = default;
+    CertificateAuthenticatedSessionEstablishment & operator=(CertificateAuthenticatedSessionEstablishment &&) = default;
+
+    virtual ~CertificateAuthenticatedSessionEstablishment();
+
+    /**
+     * @brief
+     *   Initialize using operational credentials code and wait for session establishment requests.
+     *
+     * @param myNodeId        Node id of local node
+     * @param myKeyId         Key ID to be assigned to the secure session on the peer node
+     * @param delegate        Callback object
+     *
+     * @return CHIP_ERROR     The result of initialization
+     */
+    CHIP_ERROR WaitForSessionEstablishment(NodeId myNodeId, uint16_t myKeyId, AuthenticatedSessionEstablishmentDelegate * delegate);
+
+    /**
+     * @brief
+     *   Create and send session establishment request using device's operational credentials.
+     *
+     * @param peerAddress      Address of peer to pair
+     * @param myNodeId         Node id of local node
+     * @param peerNodeId       Node id of the peer node
+     * @param myKeyId          Key ID to be assigned to the secure session on the peer node
+     * @param delegate         Callback object
+     *
+     * @return CHIP_ERROR      The result of initialization
+     */
+    CHIP_ERROR EstablishSession(const Transport::PeerAddress peerAddress, NodeId myNodeId, NodeId peerNodeId, uint16_t myKeyId,
+                                AuthenticatedSessionEstablishmentDelegate * delegate);
+
+    /**
+     * @brief
+     *   Derive a secure session from the paired session. The API will return error
+     *   if called before pairing is established.
+     *
+     * @param info        Information string used for key derivation
```

Comment:
This is pretty unclear.  Where does this come from?  Needs more details...

Reply:
It's a fixed set of strings, as per spec. I think we can remove it from the public API, since the key generated for the session is always using a specific string from the set.